R/fit.R

```diff
@@ -273,6 +273,36 @@ CmdStanMCMC <- R6::R6Class(
       } else {
         cat(paste(self$runset$procs$chain_output(id), collapse="\n"))
       }
+    },
+
+    draws = function(inc_warmup = FALSE) {
+      if (!length(self$output_files())) {
+        stop("No chains finished successfully. Unable to retrieve the fit.",
+             call. = FALSE)
+      }
+      if (is.null(private$draws_)) {
+        private$read_csv_()
+      }
+      if(inc_warmup && private$sampling_info_$save_warmup) {
```

Comment:
Do we want to warn users if they use inc_warmup = TRUE but save_warmup was false for the sampling?

Reply:
Yeah, at least a warning. Actually maybe an error makes more sense. But definitely either a warning or error

Reply:
Done. Added errors. Please let me know if the error messages need improving.

Reply:
> There are 2 more open questions here:
> 
> * where does lp__ belong? Its currently in draws()
> * names

For `lp__` I think I slightly prefer it to be accessed via `fit$sampler_diagnostics()` rather than `fit$draws()` but I'm not 100% sure. @bbbales2 @mitzimorris what do you think?

@rok-cesnovar For names, do you mean the names of the methods?

Reply:
Actually nevermind on moving `lp__`. let's keep it in `draws()` so that it's included in the `$summary()` output.